domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/service/ReviewService.kt

```diff
@@ -31,4 +34,71 @@ class ReviewService(
    */
   fun getCompletedReviewsForPrisoner(prisonNumber: String): List<CompletedReview> =
     reviewPersistenceAdapter.getCompletedReviews(prisonNumber)
+
+  fun determineReviewScheduleCalculationRule(releaseDate: LocalDate?, sentenceType: SentenceType, isReAdmission: Boolean, isTransfer: Boolean): ReviewScheduleCalculationRule =
+    if (isReAdmission) {
+      ReviewScheduleCalculationRule.PRISONER_READMISSION
+    } else if (isTransfer) {
+      ReviewScheduleCalculationRule.PRISONER_TRANSFER
+    } else {
+      when (sentenceType) {
+        SentenceType.REMAND -> ReviewScheduleCalculationRule.PRISONER_ON_REMAND
+        SentenceType.CONVICTED_UNSENTENCED -> ReviewScheduleCalculationRule.PRISONER_UN_SENTENCED
+        SentenceType.INDETERMINATE_SENTENCE -> ReviewScheduleCalculationRule.INDETERMINATE_SENTENCE
+
+        else -> reviewScheduleCalculationRuleBasedOnTimeLeftToServe(releaseDate!!)
+      }
+    }
+
+  fun calculateReviewWindow(reviewScheduleCalculationRule: ReviewScheduleCalculationRule): ReviewScheduleWindow =
+    when (reviewScheduleCalculationRule) {
+      ReviewScheduleCalculationRule.PRISONER_READMISSION, ReviewScheduleCalculationRule.PRISONER_TRANSFER -> ReviewScheduleWindow.fromTodayToTenDays()
+      ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE -> ReviewScheduleWindow.fromOneToThreeMonths()
+      ReviewScheduleCalculationRule.BETWEEN_6_AND_12_MONTHS_TO_SERVE, ReviewScheduleCalculationRule.PRISONER_ON_REMAND, ReviewScheduleCalculationRule.PRISONER_UN_SENTENCED -> ReviewScheduleWindow.fromTwoToThreeMonths()
+      ReviewScheduleCalculationRule.BETWEEN_12_AND_60_MONTHS_TO_SERVE -> ReviewScheduleWindow.fromFourToSixMonths()
+      ReviewScheduleCalculationRule.MORE_THAN_60_MONTHS_TO_SERVE, ReviewScheduleCalculationRule.INDETERMINATE_SENTENCE -> ReviewScheduleWindow.fromTenToTwelveMonths()
+    }
+
+  private fun reviewScheduleCalculationRuleBasedOnTimeLeftToServe(releaseDate: LocalDate): ReviewScheduleCalculationRule {
+    val today = LocalDate.now()
+    val timeLeftToServe = Period.between(today, releaseDate)
+    val timeLeftToServeInMonths = timeLeftToServe.toTotalMonths()
+    val timeLeftToServeRemainderDays = timeLeftToServe.minusMonths(timeLeftToServeInMonths).days
+
+    return if (timeLeftToServeInMonths < 6 || (timeLeftToServeInMonths == 6L && timeLeftToServeRemainderDays == 0)) {
+      ReviewScheduleCalculationRule.LESS_THAN_6_MONTHS_TO_SERVE
+    } else if (timeLeftToServeInMonths < 12 || (timeLeftToServeInMonths == 12L && timeLeftToServeRemainderDays == 0)) {
+      ReviewScheduleCalculationRule.BETWEEN_6_AND_12_MONTHS_TO_SERVE
+    } else if (timeLeftToServeInMonths < 60 || (timeLeftToServeInMonths == 60L && timeLeftToServeRemainderDays == 0)) {
+      ReviewScheduleCalculationRule.BETWEEN_12_AND_60_MONTHS_TO_SERVE
+    } else {
+      ReviewScheduleCalculationRule.MORE_THAN_60_MONTHS_TO_SERVE
+    }
+  }
+}
+
+enum class SentenceType {
```

Comment:
This enum _might_ get renamed as I flesh out more code, but basically is the sentence type and is equivilent to the `legalStatus` field from the `Prisoner` record returned from `prisoner-search-api` (naming is hard!)

I might also move it out of this source file into a file of it's own (and possible a different package, not sure yet); along with also potentially moving/renaming `ReviewScheduleWindow` below (though that feels like a better name, and feels appropriately close enough to where it is being used 🤔 )